libs/partners/cohere/langchain_cohere/cohere_agent.py

```diff
@@ -11,7 +11,12 @@
 from langchain_core.runnables import Runnable, RunnablePassthrough
 from langchain_core.runnables.base import RunnableLambda
 from langchain_core.tools import BaseTool
-from langchain_core.utils.function_calling import convert_to_openai_function
+from langchain_core.utils.function_calling import (
+    PYTHON_TO_JSON_TYPES,
```

Comment:
this only handles primitive types (i.e. not `array` and `object`). see below for a more complete solution.